File: FlutterProjects/Practice/Modifying/hr_and_pms - v5 Upgrading/lib/features/bonus/model/BonusType.dart
enum BonusType {PERFORMANCE, ANNUAL, FESTIVAL, PROMOTIONAL}

extension BonusTypeExtension on BonusType {
  static BonusType fromString(String bonusTypeString) {
    return BonusType.values.firstWhere(
          (bonusType) => bonusType.toString().split('.').
          last.toUpperCase() == bonusTypeString.toUpperCase(),
    );
  }

  String toShortString() {
    return toString().split('.').last.toUpperCase();
  }
}

File: FlutterProjects/Practice/Modifying/Backend with JWT- v8 Upgrading/src/main/java/com/mrahmed/HRandPayrollManagementSystem/entity/BonusType.java
package com.mrahmed.HRandPayrollManagementSystem.entity;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum BonusType {
    PERFORMANCE, ANNUAL, FESTIVAL, PROMOTIONAL;


    @JsonCreator
    public static BonusType fromString(String value) {
        return BonusType.valueOf(value.toUpperCase());
    }

    @JsonValue
    public String toJson() {
        return this.name();
    }
}